server/event.cpp

```diff
@@ -114,6 +114,30 @@ void ServerInstance::getEventProfilingInfo()
 	mStream.write(reply);
 }
 
+void ServerInstance::setEventCallback()
+{
+	SetEventCallback packet = mStream.read<SetEventCallback>();
+	cl_event event = getObj<cl_event>(packet.mEventID);
+
+	mEventCallbacks.emplace_back(packet.mID, [this](uint32_t id) {
+		// TODO: should we send callback packets immediately by flushing the packet stream?
+		mEventStream.write<FireEventCallback>({id});
```

Comment:
FIXME: flushing the stream here crashes (as in SEGFAULTs) the client. I don't know why that happens yet.